enhancements/platform/quota-management/README.md

```diff
@@ -0,0 +1,769 @@
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+software ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside of your immediate team.
+-->
+
+Different risks must be taken into account when considering implementation of the Quota Management system to ensure the system is working as expected and the risks are mitigated.
+
+- The potential to block all resource creation, preventing administration of resources due to network failure or timeouts.
+- The potential for actual resource usage being out-of-sync with the cluster-state, leading to:
+      * Allowing allocation of resources beyond the set quota limits on both external and internal levels, bypassing enforcement.
+      - Denying resource allocation when there are enough free resources to allow the request the ability to proceed.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Custom Resource Definitions 
+
+Two main CRDs will be created as a direct part of the Quota Management implementation: `ResourceQuotaClaim` and `ResourceQuotaGrant`. Ancillary to the system, a third CRD will be created that represents billing accounts for organizations.
+
+#### `ResourceQuotaClaim`
+
+The `ResourceQuotaClaim` CRD represents the *intent* of the request to create, update/scale, or delete resources. This CRD contains the resources being requested, along with their quantity. The below `yaml` snippet shows an claim created by an incoming request tied to one Instance, which requests the additional allocation of 8 CPU cores and 32GiB of memory, along with the instance count:
+
+```yaml
+apiVersion: quota.datumapis.com/v1alpha1
+kind: ResourceQuotaClaim
+metadata:
+  # Connect the claim’s lifetime to the workload that needs the quota
+  name: instance-abc123-claim
+  namespace: proj-abc 
+  ownerReferences: 
+  - apiVersion: compute.datumapis.com/v1
+    kind: Instance
+    name: instance-abc123
+    uid: <uuid>
+  finalizers:
+  - quota.datumapis.com/usage-release
+spec:
+  # Resources being requested
+  resources:
+  - name: compute.datumapis.com/instances/cpu
+    quantity: "8”
+  - name: compute.datumapis.com/instances/memoryAllocated
+    quantity: "32GiB"
+  - name: compute.datumapis.com/instances/count
+    quantity: "1"
+
+
```

Comment:
I would recommend we have a dedicated resource reference option in the spec that can be used to control the resource that requested the resources. We should also use an unversioned resource reference that relies on the apiGroup instead of the apiVersion so this claim could still be valid if we update the version of a resource. 

```yaml
spec:
  resourceRef:
    apiGroup: compute.datumapis.com
    kind: Instance
    name: instance-abc123
    uid: <uid>
  ...
```